modules/oe_translation_local/modules/oe_translation_multivalue/oe_translation_multivalue.module

```diff
@@ -0,0 +1,264 @@
+ * we set in the widget alter.
+ *
+ * @param array $element
+ *   The element.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ * @param array $complete_form
+ *   The form.
+ *
+ * @return array
+ *   The element.
+ */
+function oe_translation_multivalue_address_process(array &$element, FormStateInterface $form_state, array &$complete_form) {
+  if (!isset($element['#default_value']['translation_id'])) {
+    return $element;
+  }
+  $element['translation_id'] = [
+    '#type' => 'hidden',
+    '#value' => $element['#default_value']['translation_id'],
+  ];
+  return $element;
+}
+
+/**
+ * Get a list of the form widgets that should be altered.
+ *
+ * These are the widgets we alter the form of in order to put a hidden
+ * translation_id form element with a random ID.
+ *
+ * @return string[]
+ *   The widget IDs.
+ */
+function oe_translation_multivalue_widgets() {
+  $field_types = oe_translation_multivalue_field_types();
+  $options = [];
+  $plugin_manager = \Drupal::service('plugin.manager.field.widget');
+  foreach ($field_types as $field_type) {
+    $options += $plugin_manager->getOptions($field_type);
+  }
+
+  return array_keys($options);
+}
+
+/**
+ * Get a list of the field types we are altering.
+ *
+ * @return string[]
+ *   The field IDs.
+ */
+function oe_translation_multivalue_field_types() {
+  return [
+    'link',
+    'address',
+    'description_list_field',
+    'link_description',
+    'string',
+    'timeline_field',
+    'typed_link',
+  ];
+}
+
+/**
+ * Implements hook_field_widget_complete_form_alter().
+ */
+function oe_translation_multivalue_field_widget_complete_form_alter(&$field_widget_complete_form, FormStateInterface $form_state, $context) {
+  $widgets = oe_translation_multivalue_widgets();
+  $widget = $context['widget'];
+  if (!in_array($widget->getPluginId(), $widgets)) {
+    return;
+  }
+  $items = $context['items'];
+  $property_definitions = $items->getFieldDefinition()->getItemDefinition()->getPropertyDefinitions();
+  if (!isset($property_definitions['translation_id'])) {
+    return;
+  }
+
+  $id_generator = \Drupal::service('oe_translation_multivalue.translation_id_generator');
+  $field_id = $items->getFieldDefinition()->getFieldStorageDefinition()->id();
+  $next_id = $id_generator->getNextId($field_id);
+
+  foreach ($items as $delta => $item) {
+    // If the field delta doesn't yet have a translation ID, generate the next
+    // one and set that. However, if we do generate a new one, we need to also
+    // increment it here with each iteration because each delta needs to
+    // receive a new one.
+    $translation_id = is_numeric($item->translation_id) ? $item->translation_id : NULL;
+    if (!$translation_id) {
+      $translation_id = $next_id;
+      $next_id++;
+    }
+
+    // For address, we have an exception because it uses a form element.
+    if ($widget->getPluginId() === 'address_default') {
```

Comment:
Any way to not add the exception directly here ?

Reply:
so far we only have this field type. Once we have more we can refactor i guess to move away the exceptions